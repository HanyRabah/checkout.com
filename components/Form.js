import React from 'react'
import styled from 'styled-components'
import Input from './Input'
import Rating from './Rating'
import useForm from '../hooks/useForm'
import validate from '../helpers/validate'

const Form = ({ submitComment }) => {
  const initialValues = {
    name: '',
    email: '',
    rating: '',
    comment: '',
    date: '',
  }

  const submitForm = () => {
    submitComment({ ...values })
  }

  const { values, errors, handleChange, handleSubmit } = useForm(
    initialValues,
    submitForm,
    validate,
  )

  return (
    <FormWrapper onSubmit={handleSubmit} noValidate data-testid="form">
      <FormTitle>
        <FormTitleText>Give Us Feedback</FormTitleText>
        <FormTitleTextSmall>
          Help us make our service better :)
        </FormTitleTextSmall>
      </FormTitle>

      <Input
        type="text"
        label="What is your name ?"
        name="name"
        placeholder="Name"
        errorText={errors.name}
        handleChange={handleChange}
        value={values.name || ''}
        required
        data-testid="name-input"
      />

      <Input
        type="text"
        label="What is your email ?"
        name="email"
        placeholder="E-mail"
        errorText={errors.email}
        handleChange={handleChange}
        value={values.email || ''}
        required
        data-testid="email-input"
      />

      <Rating
        label="Overall, how satisfiedy were you with our service ?"
        name="rating"
        errorText={errors.rating}
        handleChange={handleChange}
        value={values.rating || ''}
        required
        data-testid="rating-input"
      />

      <Input
        type="textArea"
        label="Any Feedback ?"
        name="comment"
        placeholder="Add your feedback"
        errorText={errors.comment}
        handleChange={handleChange}
        value={values.comment || ''}
        row={3}
        required
        data-testid="textarea-input"
      />

      <Button type="submit" data-testid="submit-form">
        Send My Feedback
      </Button>
    </FormWrapper>
  )
}

const FormWrapper = styled.form`
  padding: 48px;
`
const FormTitle = styled.div`
  margin-bottom: 22px;
`
const FormTitleText = styled.h1`
  margin: 0;
`
const FormTitleTextSmall = styled.small`
  color: #6c757d;
`

const Button = styled.button`
  background: #fd496d;
  color: white;
  font-weight: 500;
  border: 0;
  font-size: 18px;
  padding: 18px;
  border-radius: 6px;
  width: 100%;
  outline: none;
  transition: background 300ms ease-out;
  &:hover,
  &:focus {
    background: #ed4b8b;
    outline: none;
  }
`

export default Form
