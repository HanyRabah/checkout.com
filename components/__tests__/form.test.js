import React from 'react';
import Form from '../Form';
import renderer from 'react-test-renderer';
import { fireEvent, cleanup, render } from '@testing-library/react';


describe('Form', () => {
    afterEach(cleanup);
  
    it('render correctly', () => {
      const component = renderer.create(<Form />);
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    // test('Form Submit', () => {
    //     const onSubmit = jest.fn();
    //     const { getByTestId } = render(<Form onSubmit={onSubmit} />);
    //     fireEvent.click(getByTestId('form'));
    //     expect(onSubmit).toHaveBeenCalled();
    // })
  });

 
