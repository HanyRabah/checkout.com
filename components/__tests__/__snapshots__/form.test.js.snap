// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form render correctly 1`] = `
<form
  className="Form__FormWrapper-sc-1vmu4ff-0 iPGZti"
  data-testid="form"
  noValidate={true}
  onSubmit={[Function]}
>
  <div
    className="Form__FormTitle-sc-1vmu4ff-1 dNyNdB"
  >
    <h1
      className="Form__FormTitleText-sc-1vmu4ff-2 hJdVCJ"
    >
      Give Us Feedback
    </h1>
    <small
      className="Form__FormTitleTextSmall-sc-1vmu4ff-3 lajXBD"
    >
      Help us make our service better :)
    </small>
  </div>
  <div
    className="Input__InputWrapper-p4pzeo-0 iLsCTM"
  >
    <label
      className="Input__Label-p4pzeo-1 inQhSJ"
    >
      <span
        className="Input__Required-p4pzeo-5 lnVpvf"
        title="required"
      >
        *
      </span>
       
      What is your name ?
    </label>
    <input
      className="Input-p4pzeo-2 kEMRak"
      name="name"
      onChange={[Function]}
      placeholder="Name"
      value=""
    />
  </div>
  <div
    className="Input__InputWrapper-p4pzeo-0 iLsCTM"
  >
    <label
      className="Input__Label-p4pzeo-1 inQhSJ"
    >
      <span
        className="Input__Required-p4pzeo-5 lnVpvf"
        title="required"
      >
        *
      </span>
       
      What is your email ?
    </label>
    <input
      className="Input-p4pzeo-2 kEMRak"
      name="email"
      onChange={[Function]}
      placeholder="E-mail"
      value=""
    />
  </div>
  <div
    className="Rating__Container-sc-8fqffi-0 BzQdw"
  >
    <p
      className="Rating__Label-sc-8fqffi-1 iCfwlL"
    >
      <span
        className="Rating__Required-sc-8fqffi-7 ksrtbr"
        title="required"
      >
        *
      </span>
       
      Overall, how satisfiedy were you with our service ?
    </p>
    <div
      className="Rating__RatingContainer-sc-8fqffi-3 eMkkQo"
    >
      <div
        className="Rating__Star-sc-8fqffi-4 gIripz"
      >
        <input
          className="Rating__Radio-sc-8fqffi-5 cfQPPy"
          name="rating"
          onChange={[Function]}
          type="radio"
          value={1}
        />
        <svg
          className="Stars__Star-agauv2-0 htjMpb"
          selected={false}
          viewBox="0 0 36 34"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m23.363 8.584-7.378-1.127-3.307-7.044c-.247-.526-1.11-.526-1.357 0l-3.306 7.044-7.378 1.127c-.606.093-.848.83-.423 1.265l5.36 5.494-1.267 7.767c-.101.617.558 1.08 1.103.777l6.59-3.642 6.59 3.643c.54.3 1.205-.154 1.103-.777l-1.267-7.767 5.36-5.494c.425-.436.182-1.173-.423-1.266z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        className="Rating__Star-sc-8fqffi-4 gIripz"
      >
        <input
          className="Rating__Radio-sc-8fqffi-5 cfQPPy"
          name="rating"
          onChange={[Function]}
          type="radio"
          value={2}
        />
        <svg
          className="Stars__Star-agauv2-0 htjMpb"
          selected={false}
          viewBox="0 0 36 34"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m23.363 8.584-7.378-1.127-3.307-7.044c-.247-.526-1.11-.526-1.357 0l-3.306 7.044-7.378 1.127c-.606.093-.848.83-.423 1.265l5.36 5.494-1.267 7.767c-.101.617.558 1.08 1.103.777l6.59-3.642 6.59 3.643c.54.3 1.205-.154 1.103-.777l-1.267-7.767 5.36-5.494c.425-.436.182-1.173-.423-1.266z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        className="Rating__Star-sc-8fqffi-4 gIripz"
      >
        <input
          className="Rating__Radio-sc-8fqffi-5 cfQPPy"
          name="rating"
          onChange={[Function]}
          type="radio"
          value={3}
        />
        <svg
          className="Stars__Star-agauv2-0 htjMpb"
          selected={false}
          viewBox="0 0 36 34"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m23.363 8.584-7.378-1.127-3.307-7.044c-.247-.526-1.11-.526-1.357 0l-3.306 7.044-7.378 1.127c-.606.093-.848.83-.423 1.265l5.36 5.494-1.267 7.767c-.101.617.558 1.08 1.103.777l6.59-3.642 6.59 3.643c.54.3 1.205-.154 1.103-.777l-1.267-7.767 5.36-5.494c.425-.436.182-1.173-.423-1.266z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        className="Rating__Star-sc-8fqffi-4 gIripz"
      >
        <input
          className="Rating__Radio-sc-8fqffi-5 cfQPPy"
          name="rating"
          onChange={[Function]}
          type="radio"
          value={4}
        />
        <svg
          className="Stars__Star-agauv2-0 htjMpb"
          selected={false}
          viewBox="0 0 36 34"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m23.363 8.584-7.378-1.127-3.307-7.044c-.247-.526-1.11-.526-1.357 0l-3.306 7.044-7.378 1.127c-.606.093-.848.83-.423 1.265l5.36 5.494-1.267 7.767c-.101.617.558 1.08 1.103.777l6.59-3.642 6.59 3.643c.54.3 1.205-.154 1.103-.777l-1.267-7.767 5.36-5.494c.425-.436.182-1.173-.423-1.266z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        className="Rating__Star-sc-8fqffi-4 gIripz"
      >
        <input
          className="Rating__Radio-sc-8fqffi-5 cfQPPy"
          name="rating"
          onChange={[Function]}
          type="radio"
          value={5}
        />
        <svg
          className="Stars__Star-agauv2-0 htjMpb"
          selected={false}
          viewBox="0 0 36 34"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m23.363 8.584-7.378-1.127-3.307-7.044c-.247-.526-1.11-.526-1.357 0l-3.306 7.044-7.378 1.127c-.606.093-.848.83-.423 1.265l5.36 5.494-1.267 7.767c-.101.617.558 1.08 1.103.777l6.59-3.642 6.59 3.643c.54.3 1.205-.154 1.103-.777l-1.267-7.767 5.36-5.494c.425-.436.182-1.173-.423-1.266z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
  </div>
  <div
    className="Input__InputWrapper-p4pzeo-0 iLsCTM"
  >
    <label
      className="Input__Label-p4pzeo-1 inQhSJ"
    >
      <span
        className="Input__Required-p4pzeo-5 lnVpvf"
        title="required"
      >
        *
      </span>
       
      Any Feedback ?
    </label>
    <textarea
      className="Input__TextArea-p4pzeo-3 jXbrWD"
      name="comment"
      onChange={[Function]}
      placeholder="Add your feedback"
      rows={3}
      value=""
    />
  </div>
  <button
    className="Form__Button-sc-1vmu4ff-4 cLmawq"
    data-testid="submit-form"
    type="submit"
  >
    Send My Feedback
  </button>
</form>
`;
